x-pack/spec/config_management/elasticsearch_source_spec.rb

```diff
@@ -216,267 +287,312 @@
     let(:pipeline_id) { "apache" }
     let(:mock_client)  { double("http_client") }
     let(:settings) { super.merge({ "xpack.management.pipeline.id" => pipeline_id }) }
-    let(:es_path) { "#{described_class::PIPELINE_INDEX}/_mget" }
+    let(:config) { "input { generator {} } filter { mutate {} } output { }" }
+    let(:elasticsearch_response) { elasticsearch_8_response }
+    let(:elasticsearch_8_response) {
+      "{\"#{pipeline_id}\":{
+          \"username\":\"log.stash\",
+          \"modified_timestamp\":\"2017-02-28T23:02:17.023Z\",
+          \"pipeline_metadata\":{\"version\":5,\"type\":\"logstash_pipeline\"},
+          \"pipeline\":\"#{config}\",
+          \"pipeline_settings\":{\"pipeline.batch.delay\":\"50\", \"pipeline.workers\":\"99\", \"pipeline.output.workers\":\"99\",  \"nonsensical.invalid.setting\":\"-9999\"}}}" }
+
+    let(:elasticsearch_7_9_response) {
+      "{ \"docs\":[{
+          \"_index\":\".logstash\",
+          \"_type\":\"pipelines\",
+          \"_id\":\"#{pipeline_id}\",
+          \"_version\":8,
+          \"found\":true,
+          \"_source\":{
+              \"id\":\"apache\",
+              \"description\":\"Process apache logs\",
+              \"modified_timestamp\":\"2017-02-28T23:02:17.023Z\",
+              \"pipeline_metadata\":{\"version\":5,\"type\":\"logstash_pipeline\",\"username\":\"elastic\"},
+              \"pipeline\":\"#{config}\",
+              \"pipeline_settings\":{\"pipeline.workers\":\"99\", \"pipeline.output.workers\":\"99\",  \"nonsensical.invalid.setting\":\"-9999\"}}}]}" }
+    let(:es_path) { ".logstash/_mget" }
     let(:request_body_string) { LogStash::Json.dump({ "docs" => [{ "_id" => pipeline_id }] }) }
 
     before do
-      allow(mock_client).to receive(:post).with(es_path, {}, request_body_string).and_return(LogStash::Json.load(elasticsearch_response))
+      allow(mock_client).to receive(:get).with(system_indices_url_regex).and_return(LogStash::Json.load(elasticsearch_response))
+      allow(mock_client).to receive(:get).with("/").and_return(es_version_response)
+      allow(mock_client).to receive(:post).with(es_path, {}, request_body_string).and_return(LogStash::Json.load(elasticsearch_7_9_response))
       allow(mock_license_client).to receive(:get).with('_xpack').and_return(valid_xpack_response)
       allow_any_instance_of(LogStash::LicenseChecker::LicenseReader).to receive(:client).and_return(mock_license_client)
-
     end
 
-    context "with one `pipeline_id` configured" do
-      context "when successfully fetching a remote configuration" do
+    describe "system indices [8] and legacy api [7.9]" do
+      [8, 7.9].each { |es_version|
```

Comment:
Test the new flow and the old flow to make sure backward compatibility

Reply:
All test cases are kept. Most of the changes are for simplicity, readability, and the ease of testing both versions